### ISSUE TITLE
Compiling on OSX

### ISSUE THREAD

**Issue by dennisguse** [TheTelephone/TheTelephone#2]
Although CMake is used for building the project, it has only been tested on Linux.
It should work on OSX almost by default, but this needs verification.
Additional required steps should be added to README.md

**Comment by dennisguse** [comment]
I started working towards compiling on MACOS and made some progress, which is available in branch: [MACOS](https://github.com/TheTelephone/TheTelephone/tree/MACOS).

It uses [Homebrew](brew.sh) and contains:
- Install instructions in README.md
- CMake configuration adjusted (for Clang)

**Status**: Works for all externals not depending on libraries.
For some reason, clang fails to find the homebrew installed libraries, and I could not figure out why.

**Disclaimer**: I adjusted the CMake configuration from what I remembered doing 5 hours earlier.
It should work, but consider untested.

PS: I don't have a MAC available for further investigation.

**Comment by chohner** [comment]
It seems to work by replacing CHECK_LIBRARY_EXISTS with find_library, like so:

```
find_library(HAVE_FFTW fftw3)
find_library(HAVE_GSM gsm)
find_library(HAVE_OPUS opus)
find_library(HAVE_SNDFILE sndfile)
find_library(HAVE_SPEEX speex)
find_library(HAVE_JSON json-c)
find_library(HAVE_WEBSOCKETS websockets)
```

Also note that `json` is `json-c` on OSX. Corresponding pull request https://github.com/TheTelephone/TheTelephone/pull/7

**Comment by dennisguse** [comment]
Changes merged and also upgraded to Ubuntu 16.10 (libjson-c is now called libjson-c).
